src/nlpia/book_parser.py

```diff
@@ -0,0 +1,97 @@
+import glob
+import os
+
+
+def get_lines(file_path):
+    '''
+    args:
+        file_path  string  aboslute path of the Manning manuscript asciidoc files
+                           i.e.: /Users/cole-home/repos/nlpinaction/manuscript
+
+    returns:
+        list of strings
+    '''
+    path = os.path.join(file_path, 'Chapter*')
+    files = glob.glob(path)
+    lines = []
+    for file in files:
+        with open(file, 'r') as f:
+            lines += f.readlines()
+
+    path = os.path.join(file_path, 'Appendix*')
+    files = glob.glob(path)
+    for file in files:
+        with open(file, 'r') as f:
+            lines += f.readlines()
+    return lines
+
+
+def tag_lines(lines):
+    '''
+    Mostly naively tags lines from manuscript with:
+        natural
+        header
+        image_link
+        caption
+        code
+
+    retuns:
+        list of tuples  [(tag, line), ...]
+    '''
+    current_block_type = None
+    open_block = False
+    block_terminator = None
+    block_start = 0
+    tup_lines = []
+    for idx, line in enumerate(lines):
+
+        if line.startswith('/') or line.startswith(':'):
+            continue
+
+        if line.startswith('[source'):
```

Comment:
`line.lower().strip().replace(" ", "").startswith('[source')` may help a little. I'll add that and other refinements.